File: 프로그래머스/lv2/12911. 다음 큰 숫자/다음 큰 숫자.py
def count(n):
    result = ''
    while n > 1:
        result = str(n % 2) + result
        n //= 2 
    result = str(n) + result
    return result.count('1')

def solution(n):
    answer = 0
    tmp = count(n)
    while True:
        n += 1
        if tmp == count(n):
            answer = n
            break
    return answer

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.swift
import Foundation

func solution(_ n:Int) -> Int
{
    func countOnes(_ num: Int) -> Int {
        var count = 0
        var num = num
        while num > 0 {
            if num % 2 == 1 {
                count += 1
            }
            num /= 2
        }
        return count
    }
    
    let targetCount = countOnes(n)
    var nextNumber = n + 1
    
    while countOnes(nextNumber) != targetCount {
        nextNumber += 1
    }
    
    return nextNumber
}